src/components/PostDetails.tsx

```diff
@@ -1,107 +1,126 @@
+    }
+  };
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="block">
+        <h2 data-cy="PostTitle">{title}</h2>
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
+        <p data-cy="PostBody">{post.body}</p>
+      </div>
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+      <div className="block">
+        {isLoading ? (
+          <Loader />
+        ) : (
+          <>
+            {isError ? (
+              <div className="notification is-danger" data-cy="CommentsError">
+                Something went wrong
+              </div>
+            ) : (
+              <>
+                {comments.length === 0 ? (
+                  <p className="title is-4" data-cy="NoCommentsMessage">
+                    No comments yet
+                  </p>
+                ) : (
+                  <>
+                    <p className="title is-4">Comments:</p>
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+                    {comments.map(comment => (
+                      <article
+                        className="message is-small"
+                        data-cy="Comment"
+                        key={comment.id}
+                      >
+                        <div className="message-header">
+                          <a
+                            href={`mailto:${comment.email}`}
+                            data-cy="CommentAuthor"
+                          >
+                            {comment.name}
+                          </a>
+                          <button
+                            data-cy="CommentDelete"
+                            type="button"
+                            className="delete is-small"
+                            aria-label="delete"
+                            onClick={() => handleDeleteComment(comment)}
```

Comment:
Consider adding a confirmation step before deleting a comment to prevent accidental deletions.